modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -963,24 +939,46 @@ private Assignment generateAssignment(Experiment experiment, User.ID userID, Con
         Assignment result = builder.build();
         return assignmentsRepository.assignUser(result, experiment, date);
     }
-
+    
     /**
      * Returns a bucketList depending on whether it is a personalization experiment or not.
-     * Retrieves buckets from DE if personalization is enabled
+     * Retrieves buckets from Assignment Decorator if personalization is enabled
      * Retrieves buckets from Repository if personalization is not enabled and if skipBucketRetrieval is false
      *
      * @param experiment          Experiment
+     * @param userID              UserId
+     * @param segmentationProfile The segmentation profile
      * @param skipBucketRetrieval If false, retrieves buckets from Repository
      * @return BucketList
      */
-    protected BucketList getBucketList(Experiment experiment, Boolean skipBucketRetrieval) {
-        BucketList buckets = null;
-        if (!skipBucketRetrieval) {
-            buckets = repository.getBuckets(experiment.getID());
+    protected BucketList getBucketList(Experiment experiment, User.ID userID, SegmentationProfile segmentationProfile, Boolean skipBucketRetrieval) {
+        
+        BucketList bucketList = null;
+        Boolean isPersonalizationEnabled = experiment.getIsPersonalizationEnabled();
+        if (isPersonalizationEnabled && !Objects.isNull(assignmentDecorator)) {
```

Comment:
Very minor: No need of this condition "!Objects.isNull(assignmentDecorator)" as assignmentDecorator should not be NULL in any case...